resdk/resources/utils.py

```diff
@@ -94,3 +96,35 @@ def endswith_colon(schema, field):
     """
     if field in schema and not schema[field].endswith(':'):
         schema[field] += ':'
+
+
+def get_resource_id(resource):
```

Comment:
Is this a typical name for such functions? If not, another idea would be to use `get_resource_or_id` to be more explicit about what it does...